server/src/internalClusterTest/java/org/opensearch/snapshots/SnapshotStatusApisIT.java

```diff
@@ -357,6 +357,25 @@ public void testSnapshotStatusOnFailedSnapshot() throws Exception {
         assertEquals(SnapshotsInProgress.State.FAILED, snapshotsStatusResponse.getSnapshots().get(0).getState());
     }
 
+    public void testSnapshotStatusOnPartialSnapshot() throws Exception {
+        final String dataNode = internalCluster().startDataOnlyNode();
+        final String repoName = "test-repo";
+        final String snapshotName = "test-snap";
+        final String indexName = "test-idx";
+        createRepository(repoName, "fs");
+        // create an index with a single shard on the data node, that will be stopped
+        createIndex(indexName, singleShardOneNode(dataNode));
+        index(indexName, "_doc", "some_doc_id", "foo", "bar");
+        logger.info("--> stopping data node before creating snapshot");
+        stopNode(dataNode);
+        startFullSnapshot(repoName, snapshotName, true).get();
+        final List<SnapshotStatus> snapshotStatus = clusterAdmin().snapshotsStatus(
+            new SnapshotsStatusRequest(repoName, new String[] { snapshotName })
+        ).actionGet().getSnapshots();
```

Comment:
This can be a new method under `AbstractSnapshotIntegTestCase` with the name `getSnapshotStatus`. We can use the same method in `testStatusApiConsistency` test.

Reply:
Hi @owaiskazi19 , Thanks for the feedback. 
I actually noticed that there was already a method getSnapshotStatus() in SnapshotStatusApisIT.java already, that returned the status of the snapshot. So I've re-used the same in both the test-cases testSnapshotStatusOnPartialSnapshot and testStatusApiConsistency.

The only additional change is I've removed the assertion of the snapshotStatus.size() = 1, as the getSnapshotStatus() method anyways returns the result of the 1st item of the list and not the complete list.